scripts/loadable_app/hello.c

```diff
@@ -0,0 +1,12 @@
+int a = 9;
+int b = 9;
+
+int x()
+{
+return 7;
+}
+int main(int argc, char **argv)
+{
+	int p = x();
+	return a + p + x();
```

Comment:
crt0.o contain some bootstrap for, at least `_entry` function that eventually calls `main`. 
Although you've specified `main` to be entry function instead `_entry`, seems like some symbol(s) from crt0.o still required with current compiler flags. 
Have you tried to compile (`gcc ... -c hello.c -o hello`) and then link (`gcc ... hello.o -o hello`)? The error should rise at link stage, there you could try various verbosity flags (`-v`, maybe something else) to deduce source of dependence on crt0. There should be some set of flags to remove it, like `-ffreestanding`, but it could be not enough.